docs/source/1.0/spec/core/behavior-traits.rst

```diff
@@ -655,3 +655,223 @@ See
+
+    * - in
+      - ``string``
+      - **Required**. The location where checksum can be supplied. Valid values
+        are:
+
+        * ``header`` - Indicates the checksum should be placed in an HTTP header.
+        * ``trailer`` - Indicates the checksum should be placed in the `chunked trailer part`_
+          of the body.
+
+        An algorithm may be supported at multiple locations. The first
+        supported ``in`` member value within the list of HttpChecksum property
+        takes precedence and SHOULD be used. See :ref:`here for more behavior
+        details <httpChecksum_trait_behavior>`.
+
+    * - name
+      - ``string``
+      - **Required**. The name used as a header or trailer name for a checksum
+        type. Recommended ABNF for name is as follows:
+
+        .. code-block:: abnf
+
+            name = lower-alpha *(["-"](lower-alpha / DIGIT))
+            lower-alpha = %x61-7A
+
+        That is, it should start with a lowercase letter followed by any
+        number of lowercase letters, digits, or non-sequential hyphens.
+
+        A member with the :ref:`httpHeader-trait` or :ref:`httpPrefixHeaders-trait`
+        MAY conflict with a ``httpChecksum`` header name, allowing a
+        checksum to be supplied directly. See :ref:`here for more behavior
+        details <httpChecksum_trait_header_conflict_behavior>`.
+
+.. _chunked trailer part: https://tools.ietf.org/html/rfc7230#section-4.1.2
+
+
+.. _httpChecksum_trait_behavior:
+
+HttpChecksum Behavior
+=====================
+
+.. rubric:: Client Behavior
+
+If the ``httpChecksum`` trait has a modeled ``request`` section, for HTTP
+request, the client MUST compute the request payload checksum, using an
+algorithm defined within the request section. The computed checksum SHOULD
+be supplied at the first supported checksum location as per the order of
+:ref:`checksum properties<checksum-property>` defined within the request
+section.
+
+If the resolved location is ``header``, client MUST put the checksum into the
+HTTP request headers. If the resolved location is ``trailer``, the client MUST
+put the checksum into the `chunked trailer part`_ of the body. The header or
+trailer name to use with an algorithm, location is defined by the ``name``
+member of the :ref:`checksum property<checksum-property>`.
+
+If the ``httpChecksum`` trait has a modeled ``response`` section, for HTTP
+response, the client MUST look for a supported checksum at all supported
+locations as per the defined checksum properties. If a checksum is found, the
+client MUST validate the received checksum value by computing the corresponding
+checksum of the received payload.
+
+.. _httpChecksum_trait_header_conflict_behavior:
+
+An operation's ``input``, ``output``, and ``error`` structures MAY contain a
+member bound to an HTTP header matching the header or trailer name defined
+within the ``httpChecksum`` trait. For an HTTP request, if customer provides value
```

Comment:
Can you frame this so it's not about customers, clients, requests, or responses, but rather just matter of fact: If the HTTP header corresponding to a checksum is set explicitly, then the implementation MUST use the explicitly set header and skip computing a payload checksum.